Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -503,14 +504,19 @@ - (void)processBatchedWatchChanges:(NSArray<FSTWatchChange *> *)changes
 
   // Update in-memory resume tokens. FSTLocalStore will update the persistent view of these when
   // applying the completed FSTRemoteEvent.
+  FSTWeakify(self);
```

Comment:
This is unnecessary. The function we're calling is not capturing the block and is therefore not meaningfully extending the life of `self`. By the time the outer function returns the block will have been released.

In general for-each style blocks don't need the weakify dance.

In contrast, the blocks submitted to the async should use the weakify dance because we don't know when the async queue is going to call the block back. The stream may have been stopped and destroyed and we don't want the block in the queue to keep the `FSTStream` alive.

Reply:
Ha. Despite my (very) limited experience with objc and strongiyf/weakify in particular, this is something I should've caught. I read the block as "its-a-block-so-must-be-async-queue-related" and didn't look any further. (shamecube)

Fixed.